### ISSUE TITLE
Interactive mode

### ISSUE THREAD

**Issue by mattytrentini** [mattytrentini/mpbuild#4]
Would be interesting to use [Textual](https://github.com/Textualize/textual) to make an interactive mode where the user can navigate through the list of ports/boards - displaying information about them - and then choose to build/clean etc.

**Comment by Josverl** [comment]
You could create a simpler UX using one of the inquirer modules without the need to go full out on textual

**Comment by mattytrentini** [comment]
True, and that's a reasonable option! Though an interactive Textual application isn't terribly complex and may be more appealing..?

**Comment by Josverl** [comment]
I found textual a bit more than what I needed, but was already underway with click and rich-click, and just needed an input/choice/dropdown. 
I spend way more time on the mpflash TUI than I expected based on my Powershell TUI experience, where a lot of this is part of the language itself.
So did not want to grow the effort any further.

**Comment by mattytrentini** [comment]
A start has been made on this, see the [interactive-build](https://github.com/mattytrentini/mpbuild/tree/interactive-mode) branch.

It was _very_ straightforward to create a GUI shell application and populate the tree on the left with the list of boards. The harder part will be neatly streaming the CLI build output to a panel...

![image](https://github.com/user-attachments/assets/716a00b8-c672-4b42-8577-f41711b594a7)
